@babel/plugin-transform-react-constant-elements not found

<!-- Love SVGR? Please consider supporting our collective: 👉  https://opencollective.com/svgr/donate -->

## 🐛 Bug Report

I'm having the same issue, that was described here https://github.com/smooth-code/svgr/issues/116. I'm having it only when I'm using my webpack config to run https://github.com/styleguidist/react-styleguidist. It could be an issue with react-styleguidist itself, but since this issue came up before, I decided to start with svgr repo first.

## To Reproduce

I just fed my webpack config to react-styleguidist, which has this section
```js
{
  test: /\.svg$/,
  issuer: {
    test: /\.js$/,
  },
  include: path.resolve(ROOT_DIR, 'client-src'),
  loader: '@svgr/webpack',
}
```

## Expected behavior

There shouldn't be errors like

```bash
Module build failed (from ./node_modules/@svgr/webpack/lib/index.js):
Error: Cannot find module '@babel/plugin-transform-react-constant-elements' from '/Users/smashercosmo/Documents/projects/netcycler'
    at Function.module.exports [as sync] (/Users/smashercosmo/Documents/projects/netcycler/node_modules/resolve/lib/sync.js:43:15)
    at resolveStandardizedName (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/files/plugins.js:101:31)
    at resolvePlugin (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/files/plugins.js:54:10)
    at loadPlugin (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/files/plugins.js:62:20)
    at createDescriptor (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/config-descriptors.js:154:9)
    at items.map (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/config-descriptors.js:109:50)
    at Array.map (<anonymous>)
    at createDescriptors (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/config-descriptors.js:109:29)
    at createPluginDescriptors (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/config-descriptors.js:105:10)
    at alias (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/config-descriptors.js:63:49)
    at cachedFunction (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/caching.js:33:19)
    at plugins.plugins (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/config-descriptors.js:28:77)
    at mergeChainOpts (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/config-chain.js:319:26)
    at /Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/config-chain.js:283:7
    at buildRootChain (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/config-chain.js:68:29)
    at loadPrivatePartialConfig (/Users/smashercosmo/Documents/projects/netcycler/node_modules/@svgr/webpack/node_modules/@babel/core/lib/config/partial.js:85:55)

```

## Link to repl or repo (highly encouraged)

I will provide demo in case nothing immediately comes to your mind)

## Run `npx envinfo --system --binaries --npmPackages @svgr/core,@svgr/cli,@svgr/webpack,@svgr/rollup --markdown --clipboard`

Paste the results here:

```bash
## System:
 - OS: macOS Sierra 10.12.6
 - CPU: (4) x64 Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz
 - Memory: 584.66 MB / 16.00 GB
 - Shell: 5.2 - /bin/zsh
## Binaries:
 - Node: 10.12.0 - /usr/local/bin/node
 - Yarn: 1.12.1 - ~/Documents/projects/netcycler/node_modules/.bin/yarn
 - npm: 6.4.1 - /usr/local/bin/npm
## npmPackages:
 - @svgr/webpack: 4.0.3 => 4.0.3 

```

P. S. Thank you for the awesome tool)

--- a/packages/rollup/src/index.js
+++ b/packages/rollup/src/index.js
@@ -1,9 +1,22 @@
 import fs from 'fs'
 import convert from '@svgr/core'
 import { createFilter } from 'rollup-pluginutils'
-import { transform as babelTransform } from '@babel/core'
+import { transform as babelTransform, createConfigItem } from '@babel/core'
 import svgo from '@svgr/plugin-svgo'
 import jsx from '@svgr/plugin-jsx'
+import presetReact from '@babel/preset-react'
+import presetEnv from '@babel/preset-env'
+import pluginTransformReactConstantElements from '@babel/plugin-transform-react-constant-elements'
+
+const babelOptions = {
+  babelrc: false,
+  configFile: false,
+  presets: [
+    createConfigItem(presetReact, { type: 'preset' }),
+    createConfigItem([presetEnv, { modules: false }], { type: 'preset' }),
+  ],
+  plugins: [createConfigItem(pluginTransformReactConstantElements)],
+}
 
 function svgrPlugin(options = {}) {
   const filter = createFilter(options.include || '**/*.svg', options.exclude)
@@ -42,21 +55,10 @@ function svgrPlugin(options = {}) {
 
       const pBabelTransform = async code =>
         new Promise((resolve, reject) => {
-          babelTransform(
-            code,
-            {
-              babelrc: false,
-              presets: [
-                '@babel/preset-react',
-                ['@babel/preset-env', { modules: false }],
-              ],
-              plugins: ['@babel/plugin-transform-react-constant-elements'],
-            },
-            (err, result) => {
-              if (err) reject(err)
-              else resolve(result.code)
-            },
-          )
+          babelTransform(code, babelOptions, (err, result) => {
+            if (err) reject(err)
+            else resolve(result.code)
+          })
         })
 
       if (babel) {

--- a/packages/webpack/src/index.js
+++ b/packages/webpack/src/index.js
@@ -1,8 +1,21 @@
 import { getOptions } from 'loader-utils'
-import { transform as babelTransform } from '@babel/core'
+import { transform as babelTransform, createConfigItem } from '@babel/core'
 import convert from '@svgr/core'
 import svgo from '@svgr/plugin-svgo'
 import jsx from '@svgr/plugin-jsx'
+import presetReact from '@babel/preset-react'
+import presetEnv from '@babel/preset-env'
+import pluginTransformReactConstantElements from '@babel/plugin-transform-react-constant-elements'
+
+const babelOptions = {
+  babelrc: false,
+  configFile: false,
+  presets: [
+    createConfigItem(presetReact, { type: 'preset' }),
+    createConfigItem([presetEnv, { modules: false }], { type: 'preset' }),
+  ],
+  plugins: [createConfigItem(pluginTransformReactConstantElements)],
+}
 
 function svgrLoader(source) {
   const callback = this.async()
@@ -25,23 +38,10 @@ function svgrLoader(source) {
 
   const pBabelTransform = async jsCode =>
     new Promise((resolve, reject) => {
-      babelTransform(
-        jsCode,
-        {
-          babelrc: false,
-          // Unless having this, babel will merge the config with global 'babel.config.js' which may causes some problems such as using react-hot-loader/babel in babel.config.js
-          configFile: false,
-          presets: [
-            '@babel/preset-react',
-            ['@babel/preset-env', { modules: false }],
-          ],
-          plugins: ['@babel/plugin-transform-react-constant-elements'],
-        },
-        (err, result) => {
-          if (err) reject(err)
-          else resolve(result.code)
-        },
-      )
+      babelTransform(jsCode, babelOptions, (err, result) => {
+        if (err) reject(err)
+        else resolve(result.code)
+      })
     })
 
   const tranformSvg = svg =>

